### PR TITLE
make dep ensure verbose, -vendor-only in CircleCI

### PR DESCRIPTION
**Pull request by rgarcia**
see https://clever.atlassian.net/wiki/spaces/ENG/pages/217481220/2017-12-15+Meeting+notes

### CODE REVIEW

File: make/golang.mk [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # This is the default Clever Golang Makefile.
 # It is stored in the dev-handbook repo, github.com/Clever/dev-handbook
 # Please do not alter this file directly.
-GOLANG_MK_VERSION := 0.2.0
+GOLANG_MK_VERSION := 0.2.1
```
Comment (natebrennand) [inline make/golang.mk]: shouldn't this be minor to convey the breaking nature of it?

**Comment by rgarcia** [comment]
tested via
```
make install_deps
```
and
```
CIRCLECI=x make install_deps
```
and noticing the difference in what command was run

File: make/golang.mk [hunk 2]
```diff
@@ -52,9 +52,16 @@ golang-dep-vendor-deps: bin/dep
 # golang-godep-vendor is a target for saving dependencies with the dep tool
 # to the vendor/ directory. All nested vendor/ directories are deleted via
 # the prune command.
+# In CI, -vendor-only is used to avoid updating the lock file.
+ifndef CIRCLECI
```
Comment (natebrennand) [inline make/golang.mk]: thoughts on using `CI` to be more generalized?

Comment (natebrennand) [reply make/golang.mk]: This _is_ provided by circle ci

File: make/golang.mk (hunk 1 above)
Comment (rgarcia) [reply make/golang.mk]: 👍 a0140b7

File: make/golang.mk (hunk 2 above)
Comment (rgarcia) [reply make/golang.mk]: 👍 a0140b7

File: make/golang.mk [hunk 3]
```diff
@@ -52,9 +52,16 @@ golang-dep-vendor-deps: bin/dep
 # golang-godep-vendor is a target for saving dependencies with the dep tool
 # to the vendor/ directory. All nested vendor/ directories are deleted via
 # the prune command.
+# In CI, -vendor-only is used to avoid updating the lock file.
+ifndef CI
```
Comment (nathanleiby) [inline make/golang.mk]: @rgarcia do we prefer this approach, or doing `if` via bash like so:

https://github.com/Clever/dev-handbook/pull/103/files

Either is fine with me, just want to be consistent.

Comment (rgarcia) [reply make/golang.mk]: no preference, and i'm fine using both approaches too

**Pull request merged**